src/hotspot/share/c1/c1_IR.cpp

```diff
@@ -307,56 +307,46 @@ void IR::eliminate_null_checks() {
 }
 
 
-static int sort_pairs(BlockPair** a, BlockPair** b) {
-  if ((*a)->from() == (*b)->from()) {
-    return (*a)->to()->block_id() - (*b)->to()->block_id();
-  } else {
-    return (*a)->from()->block_id() - (*b)->from()->block_id();
-  }
-}
-
-
 class CriticalEdgeFinder: public BlockClosure {
   BlockPairList blocks;
-  IR*       _ir;
 
  public:
-  CriticalEdgeFinder(IR* ir): _ir(ir) {}
```

Comment:
Yes, we can remove the ctor, but wouldn't it be better to keep the ctor to initialize `blocks`?

Reply:
Agree. Updated